src/TheSpace/HarbergerMarket.sol

```diff
@@ -69,120 +84,177 @@ abstract contract HarbergerMarket is Multicall {
      * @dev Create Property contract, setup attached currency contract, setup tax rate
      */
     constructor(
-        string memory propertyName_,
-        string memory propertySymbol_,
-        address currencyAddress_,
-        uint256 taxRate_,
-        uint256 totalSupply_
+        string memory propertyName,
+        string memory propertySymbol,
+        address currencyAddress
     ) {
         // initialize Property contract with current contract as market
-        property = new Property(propertyName_, propertySymbol_, address(this), totalSupply_);
+        property = new Property(propertyName, propertySymbol, address(this), totalSupply);
 
         // initialize currency contract
-        currency = ERC20(currencyAddress_);
+        currency = ERC20(currencyAddress);
```

Comment:
need interfaces to change the addresses of ERC-20 token and property?

Reply:
Yes, it can be part of `TaxConfig`

Reply:
While implementing it, I realized it is not straightforward: we need to have some form of the epoch, since after changing erc-20 token address all previous tax and ubi record will become invalid.

I think it is better if we disallow updating currency addresses.